app/services/instrumentation.rb

```diff
@@ -0,0 +1,34 @@
+module Instrumentation
+  class << self
+    def append_to_log(args = {})
+      RequestStore.store[:custom_log_items] ||= {}
+      RequestStore.store[:custom_log_items].merge!(args) unless args.blank?
+      RequestStore.store[:custom_log_items]
+    end
+    alias_method :custom_log_items, :append_to_log
```

Comment:
Would it be simpler not to use the same method both for writing and reading this data? Then `append_to_log` would not require optional arguments.